pages/settings/profile.tsx

```diff
@@ -89,16 +97,26 @@ export default function Settings(props: Props) {
   const [selectedTheme, setSelectedTheme] = useState({ value: props.user.theme });
   const [selectedTimeZone, setSelectedTimeZone] = useState({ value: props.user.timeZone });
   const [selectedWeekStartDay, setSelectedWeekStartDay] = useState({ value: props.user.weekStart });
+  const [selectedLanguage, setSelectedLanguage] = useState<OptionType>({
+    value: locale,
+    label: props.localeLabels[locale],
+  });
   const [imageSrc, setImageSrc] = useState<string>(props.user.avatar);
-
   const [hasErrors, setHasErrors] = useState(false);
   const [errorMessage, setErrorMessage] = useState("");
 
+  const { i18n } = useTranslation("settings-profile-page");
+
+  useEffect(() => {
+    (async () => await i18n.changeLanguage(locale))();
+  }, [i18n, locale]);
```

Comment:
Third time I see this now - feels like it should be in a custom hook or helper

Reply:
agree. this looks overly redundant @mihaic195